src/tests/sqlsmith/src/expr.rs

```diff
@@ -31,35 +34,63 @@ lazy_static::lazy_static! {
     };
 }
 
+lazy_static::lazy_static! {
+    static ref AGG_FUNC_TABLE: HashMap<DataTypeName, Vec<AggFuncSign>> = {
+        init_agg_table()
+    };
+}
+
 fn init_op_table() -> HashMap<DataTypeName, Vec<FuncSign>> {
     let mut funcs = HashMap::<DataTypeName, Vec<FuncSign>>::new();
     func_sigs().for_each(|func| funcs.entry(func.ret_type).or_default().push(func.clone()));
     funcs
 }
 
+fn init_agg_table() -> HashMap<DataTypeName, Vec<AggFuncSign>> {
+    let mut funcs = HashMap::<DataTypeName, Vec<AggFuncSign>>::new();
+    agg_func_sigs().for_each(|func| funcs.entry(func.ret_type).or_default().push(func.clone()));
+    funcs
+}
+
 impl<'a, R: Rng> SqlGenerator<'a, R> {
-    pub(crate) fn gen_expr(&mut self, typ: DataTypeName) -> Expr {
+    pub(crate) fn gen_expr(&mut self, typ: DataTypeName, can_agg: bool, inside_agg: bool) -> Expr {
         if !self.can_recurse() {
             // Stop recursion with a simple scalar or column.
             return match self.rng.gen_bool(0.5) {
                 true => self.gen_simple_scalar(typ),
-                false => self.gen_col(typ),
+                false => self.gen_col(typ, inside_agg),
             };
         }
-        match self.rng.gen_range(0..=99) {
-            0..=99 => self.gen_func(typ),
+        assert!(!(!can_agg & inside_agg));
```

Comment:
Not very readable. Perhaps add a comment:
```suggestion
        // `gen_expr` should be generating aggregates
        // or else it should not be generating inside_agg exprs.
        assert!(can_agg || !inside_agg));
```